ng-dev/utils/ng-dev-service.ts

```diff
@@ -30,49 +32,56 @@ const firebaseConfig = {
+export async function useNgDevService<T>(argv: Argv<T>): Promise<Argv<T>>;
+export async function useNgDevService<T>(argv: Argv<T>, isAuthCommand: true): Promise<Argv<T>>;
+export async function useNgDevService<T>(
   argv: Argv<T>,
   isAuthCommand: boolean = false,
-): Argv<T> {
+): Promise<Argv<T>> {
+  const {github} = await getConfig([assertValidGithubConfig]);
+
+  if (github.useNgDevService !== true) {
+    return argv;
+  }
+
   return (
-    argv
-      .option('use-auth-service' as 'useAuthService', {
+    addGithubTokenOption(argv)
+      // TODO(josephperrott): remove once stability is validated.
+      .option('github-escape-hatch' as 'githubEscapeHatch', {
         type: 'boolean',
         default: false,
+        hidden: true,
       })
-      // TODO(josephperrott): Show flags once this tooling is rolled out.
-      .hide('use-auth-service')
-      .middleware(async (args: Arguments<T & {useAuthService: boolean}>) => {
-        initializeApp(firebaseConfig);
-        await restoreNgTokenFromDiskIfValid();
+      .middleware(
+        async (args: Arguments<T & {githubToken: string | null; githubEscapeHatch: boolean}>) => {
+          initializeApp(firebaseConfig);
+          await restoreNgTokenFromDiskIfValid();
 
-        if (args.useAuthService === false) {
-          return;
-        }
-        args.githubToken = null;
+          if (args.githubEscapeHatch === true) {
+            Log.warn('This escape hatch should only be used if the service is erroring. Please');
+            Log.warn(
+              'inform #dev-infra of the need to use this escape hatch so it can be triaged.',
+            );
+            return;
+          }
+          args.githubToken = null;
 
-        if (isAuthCommand) {
-          Log.debug('Skipping ng-dev token request as this is an auth command');
-          return;
-        }
+          if (isAuthCommand) {
+            Log.debug('Skipping ng-dev token request as this is an auth command');
+            return;
+          }
 
-        if (await getCurrentUser()) {
-          await configureAuthorizedGitClientWithTemporaryToken();
-          Log.debug('Logged into github using temporary token');
-          return;
-        }
+          if (await getCurrentUser()) {
+            await configureAuthorizedGitClientWithTemporaryToken();
+            Log.debug('Logged into github using temporary token');
+            return;
+          }
 
-        Log.error('  ✘  You must be logged in to run this command\n');
-        Log.log('Log in by running the following command:');
-        Log.log('  yarn ng-dev auth login');
-        argv.exit(1, new Error('The user is not logged in'));
-      })
+          Log.error('  ✘  You must be logged in to run this command\n');
+          Log.log('Log in by running the following command:');
+          Log.log('  yarn ng-dev auth login');
+          argv.exit(1, new Error('The user is not logged in'));
```

Comment:
I don't think we use `argv.exit` anywhere else. Can we just throw the error like elsewhere and Yargs will pick it up?

Reply:
Done.  I will come back to how we handle exiting as we currently are very inconsistent in a follow up.